aws/request/request.go

```diff
@@ -115,14 +107,14 @@ func (r *Request) WillRetry() bool {
 // and the parameters are valid. False is returned if no parameters are
 // provided or invalid.
 func (r *Request) ParamsFilled() bool {
-	return r.Params != nil && reflect.ValueOf(r.Params).Elem().IsValid()
+	return r.Params != nil
 }
 
 // DataFilled returns true if the request's data for response deserialization
 // target has been set and is a valid. False is returned if data is not
 // set, or is invalid.
 func (r *Request) DataFilled() bool {
-	return r.Data != nil && reflect.ValueOf(r.Data).Elem().IsValid()
+	return r.Data != nil
```

Comment:
The check for nil of `Param` and `Data` were to insure that a interface with a nil type value wasn't provided. Since `r.Param != nil` may fail with a false positive in the situation where a nil type value was set to the interface [[1]](https://groups.google.com/forum/#!topic/golang-nuts/QzVDKv7p0Vs) [[2]](https://golang.org/doc/faq#nil_error).
Though in practice this could only happen if a user where to set the value of the `r.Param` or `r.Data` them selves since all operation's should always provided a valid value for two fields.

Are you seeing these two calls impacting the performance of the SDK? In the pprof's we've been running this hasn't registered.